MKS CANable V2.0 (#81366)
zephyrproject-rtos/zephyr: boards/makerbase/mks_canable_v20/doc/index.rst

+-----------+------------+-------------------------------------+
| GPIO      | on-chip    | gpio                                |
+-----------+------------+-------------------------------------+
| FDCAN     | on-chip    | can                                 |
+-----------+------------+-------------------------------------+

Other hardware features are not yet supported on this Zephyr port.

The default configuration can be found in the defconfig file:
:zephyr_file:`boards/makerbase/mks_canable_v20/mks_canable_v20_defconfig`

Connections and IOs
===================

Default Zephyr Peripheral Mapping:
----------------------------------

.. rst-class:: rst-columns

- CAN_RX/BOOT0 : PB8
- CAN_TX : PB9
- D2 : PA15
- D3 : PA0
- USB_DN : PA11
- USB_DP : PA12
- SWDIO : PA13
- SWCLK : PA14
- NRST : PG10

For more details please refer to `MKS CANable V2.0 schematic`_.

System Clock
------------

The MKS CANable V2.0 system clock is driven by internal high speed oscillator.
By default system clock is driven by PLL clock at 170 MHz,
the PLL is driven by the 16 MHz high speed internal oscillator.

The FDCAN1 peripheral is driven by PLLQ, which has 170 MHz frequency sam as main

> system clock.

Author: > The CAN FD peripheral should not run at 170 MHz. Please aim for a CAN core frequency of 80 MHz for interoperability reasons (recommended by CiA).
> 

This is a good point. I see it in CiA 601-2 CAN  and here: [CiA 601-2 CAN ](https://www.can-cia.org/fileadmin/cia/documents/publications/cnlm/march_2018/18-1_p28_recommendation_for_the_canfd_bit-timing_holger_zeltwanger_cia.pdf)

Unfortunatel I can not find other way than setting main clock to 160MHz s.t. CAN clock will be 80MHz. But since the board is specifically aimed for CAN it would be totally fine.

> Also, what does "sam" mean in the sentence above?

Shall be "same". Corrected.

.. zephyr:board:: mks_canable_v20

Overview
********

> The Makerbase MKS CANable V2.0 board features an ARM Cortex-M4 based STM32G431C8 MCU

Reviewer: ```suggestion
The MakerBase MKS CANable V2.0 board features an ARM Cortex-M4 based STM32G431C8 MCU
```

> with a wide range of connectivity support and configurations.

Reviewer: A range of of? Not on this board, so I think that can be removed from the board description.

Author: MCU supports `a range of...`. But in terms of board I agree and will update

More information about the board can be found at the `MKS CANable V2.0 website`_.
It is very advisable to take a look in on user manual `MKS CANable V2.0 User Manual`_ and
schematic `MKS CANable V2.0 schematic`_ before start.

More information about STM32G431KB can be found here:

- `STM32G431C8 on www.st.com`_
- `STM32G4 reference manual`_

Supported Features
==================

The Zephyr ``mks_canable_v20`` board target supports the following hardware features:

+-----------+------------+-------------------------------------+
| Interface | Controller | Driver/Component                    |
+===========+============+=====================================+
| NVIC      | on-chip    | nested vector interrupt controller  |
+-----------+------------+-------------------------------------+
| USB       | on-chip    | universal-serial-bus                |
+-----------+------------+-------------------------------------+
| PINMUX    | on-chip    | pinmux                              |
+-----------+------------+-------------------------------------+
| GPIO      | on-chip    | gpio                                |
+-----------+------------+-------------------------------------+

> | PWM       | on-chip    | pwm                                 |

Reviewer: Where is PWM used on this board?

Author: Planned to use it for PWM LED since i see LEDs on th board as way to bright. But for the current PR it may be removed - agree.

+-----------+------------+-------------------------------------+
| FDCAN     | on-chip    | can                                 |
+-----------+------------+-------------------------------------+

Other hardware features are not yet supported on this Zephyr port.

The default configuration can be found in the defconfig file:
:zephyr_file:`boards/makerbase/mks_canable_v20/mks_canable_v20_defconfig`

Connections and IOs
===================

The stm32g431c8 mcu is the core of MKS CANable V2.0 board and it has 6 GPIO controllers.

> These controllers are responsible for pin muxing, input/output, pull-up, etc.

Reviewer: This is not that important for the board configuration and could be removed.

For more details please refer to `MKS CANable V2.0 schematic`_.

> Default Zephyr Peripheral Mapping is fixed by `MKS CANable V2.0 schematic`_:

Reviewer: What does this mean?

Author: Default Zephyr Peripheral Mapping.

----------------------------------------------------------------------------

.. rst-class:: rst-columns

- CAN_RX/BOOT0 : PB8
- CAN_TX : PB9
- D2 : PA15
- D3 : PA0
- USB_DN : PA11
- USB_DP : PA12
- SWDIO : PA13
- SWCLK : PA14
- NRST : PG10

System Clock
------------

The MKS CANable V2.0 system clock is driven by internal high speed oscillator.
By default system clock is driven by PLL clock at 170 MHz,

> the PLL is driven by the 16 MHz high speed internal oscillator.

Reviewer: At which frequency are you clocking the CAN controllers?

Author: This part describes main MCU clock and not a CAN clock. 
I will add a CAN clock description.

Programming and Debugging
*************************

MKS CANable V2.0 board includes an SWDIO debug connector header J4.

.. note::

   The debugger is not the part of the board!

Applications for the ``mks_canable_v20`` board target can be built and
flashed in the usual way (see :ref:`build_an_application` and
:ref:`application_run` for more details).

Flashing
========

> The board is configured to be flashed using west OpenOCD runner.

Reviewer: We usually don't mention this in the board documentation, as other runners can be added and used as well.

Author: Changed to: The board could be flashed using west.

Flashing an application to MKS CANable V2.0
-------------------------------------------

The debugger shall be wired to MKS CANable V2.0 board's J4 connector
according `MKS CANable V2.0 schematic`_.

Build and flash an application. Here is an example for
:zephyr:code-sample:`hello_world`.

.. zephyr-app-commands::
   :zephyr-app: samples/hello_world
   :board: mks_canable_v20
   :goals: build flash
   :west-args: -S rtt-console
   :compact:

The argument ``-S rtt-console`` is needed for debug purposes with SEGGER RTT protocol.
This option is optional and may be omitted. Omitting it frees up RAM space but prevents RTT usage.

If option ``-S rtt-console`` is selected, the connection to the target can be established as follows:

.. code-block:: console

   $ telnet localhost 9090

You should see the following message on the console:

.. code-block:: console

   $ Hello World! mks_canable_v20/stm32g431xx

.. note::

>    Current OpenOCD config will skip Segger RTT for OpenOCD under 0.12.0.

Reviewer: I would remove all of this and just refer to the common documentation on how to use Segger RTT.

Author: Somehow I can't find proper compact description for segger RTT usage same way it is done in my current documentation. Would you please put a link here?

Author: This one shows only KConfig entry and no step-by-step guide like i my documentation.

Reviewer: Feel free to open a PR for improving the documentation on using Segger RTT console, but let's make it generic instead of board-specific (as there is really nothing board-specific to it).

Author: > Feel free to open a PR for improving the documentation on using Segger RTT console, but let's make it generic instead of board-specific (as there is really nothing board-specific to it).

Definitely I could take over with my current experience. Would you please suggest a documentation section (please a link to some index.rst file), where I can place the corresponding documentation?

.. zephyr:board:: mks_canable_v20

Overview
********

> The Makervase MKS CANable V2.0 board features an ARM Cortex-M4 based STM32G431C8 MCU

Reviewer 2: ```suggestion
The Makerbase MKS CANable V2.0 board features an ARM Cortex-M4 based STM32G431C8 MCU
```

with a wide range of connectivity support and configurations.
Here are some highlights of the MKS CANable V2.0 board:

- STM32 microcontroller in LQFP48 package
- USB Type-C connector (J1)
- CAN-Bus connector (J2)
- ST-LINK/V3E debugger/programmer header (J4)
- USB VBUS power supply (5 V)
- Three LEDs: red/power_led (D1), blue/stat_led (D2), green/word_led (D3)
- One push-button for RESET

The LED red/power_led (D1) is connected directly to on-board 3.3 V and not controllable by the MCU.

More information about the board can be found at the `MKS CANable V2.0 website`_.
It is very advisable to take a look in on user manual `MKS CANable V2.0 User Manual`_ and
schematic `MKS CANable V2.0 schematic`_ before start.

> - Development support: serial wire debug (SWD), JTAG, Embedded Trace Macrocell.

Reviewer 2: This bullet point looks isolated. Should it be grouped in above list ? Or maybe rephrased ?

- USB_DN : PA11
- USB_DP : PA12
- SWDIO : PA13
- SWCLK : PA14
- NRST : PG10

System Clock
------------

The MKS CANable V2.0 system clock is driven by internal high speed oscillator.
By default system clock is driven by PLL clock at 170 MHz,
the PLL is driven by the 16 MHz high speed internal oscillator.

Programming and Debugging
*************************

MKS CANable V2.0 board includes an SWDIO debug connector header J4.

.. note::

   The debugger is not the part of the board!

Applications for the ``mks_canable_v20`` board target can be built and
flashed in the usual way (see :ref:`build_an_application` and
:ref:`application_run` for more details).

Flashing
========

The board is configured to be flashed using west OpenOCD runner.

Flashing an application to MKS CANable V2.0
-------------------------------------------

The debugger shall be wired to MKS CANable V2.0 board's J4 connector
according `MKS CANable V2.0 schematic`_.

Build and flash an application. Here is an example for
:zephyr:code-sample:`hello_world`.

.. zephyr-app-commands::
   :zephyr-app: samples/hello_world
   :board: mks_canable_v20
   :goals: build flash
   :west-args: -S rtt-console
   :compact:

> The argument ``-S rtt-console`` is needed for debug purposes with SEGGER RTT protoc`ol.

Reviewer 2: Typo ?
```suggestion
The argument ``-S rtt-console`` is needed for debug purposes with SEGGER RTT protocol.
```

> This option may is optional and may be omitted. Omitting it makes more RAM free,
> but drops RTT support.

Reviewer 2: ```suggestion
This option is optional and may be omitted. Omitting it frees up RAM space but prevents RTT usage.
```

> If option ``-S rtt-console`` were chosen, the connection to the target can be
> established as follows:

Reviewer 2: ```suggestion
If option ``-S rtt-console`` is selected, the connection to the target can be established as follows:
```
